app/javascript/src/LoginPage/loginForms/SignupForm.jsx

```diff
@@ -0,0 +1,170 @@
+// @flow
+
+import React, { Component } from 'react'
+import {
+  Form,
+  ActionGroup,
+  Button
+} from '@patternfly/react-core'
+import {
+  HiddenInputs,
+  FormGroup
+} from 'LoginPage'
+
+type Props = {
+  path: string
+}
+
+type State = {
+  username: string,
+  emailAddress: string,
+  password: string,
+  passwordConfirmation: string,
+  isValidUsername: ?boolean,
+  isValidEmailAddress: ?boolean,
+  isValidPassword: ?boolean,
+  isValidPasswordConfirmation: ?boolean
+}
+type HandlersProps = {
+  handleTextInputUsername: (username: string) => void,
+  handleTextInputEmail: (emailAddress: string) => void,
+  handleTextInputPassword: (password: string) => void,
+  handleTextInputPasswordConfirmation: (passwordConfirmation: string) => void
+}
+
+const FormGroups = ({state, handlers}: {state: State, handlers: HandlersProps}) => {
```

Comment:
I'd keep each group separated, and not sharing the state, but feeding it props only.